ctru-rs/src/services/ndsp/mod.rs

```diff
@@ -0,0 +1,310 @@
+//! NDSP (Audio) service
+
+pub mod wave;
+use wave::{WaveInfo, WaveStatus};
+
+use crate::error::ResultCode;
+use crate::services::ServiceReference;
+
+use std::cell::{RefCell, RefMut};
+use std::error;
+use std::fmt;
+use std::sync::Mutex;
+
+const NUMBER_OF_CHANNELS: u8 = 24;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum OutputMode {
+    Mono = ctru_sys::NDSP_OUTPUT_MONO,
+    Stereo = ctru_sys::NDSP_OUTPUT_STEREO,
+    Surround = ctru_sys::NDSP_OUTPUT_SURROUND,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum AudioFormat {
+    PCM8Mono = ctru_sys::NDSP_FORMAT_MONO_PCM8,
+    PCM16Mono = ctru_sys::NDSP_FORMAT_MONO_PCM16,
+    PCM8Stereo = ctru_sys::NDSP_FORMAT_STEREO_PCM8,
+    PCM16Stereo = ctru_sys::NDSP_FORMAT_STEREO_PCM16,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum InterpolationType {
+    Polyphase = ctru_sys::NDSP_INTERP_POLYPHASE,
+    Linear = ctru_sys::NDSP_INTERP_LINEAR,
+    None = ctru_sys::NDSP_INTERP_NONE,
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum NdspError {
+    InvalidChannel(u8),      // channel id
```

Comment:
Maybe documenting that the values here are channel IDs for users of the library would be a good idea
```suggestion
    /// Channel ID
    InvalidChannel(u8),
```

or something like
```rust
/// The values contained inside the members of this enum are the Channel IDs
pub enum NdspError {
...
```